### ISSUE TITLE
Provider-specific implementation to handle `Peering` configuration on API-level `Network` objects

### ISSUE THREAD

**Issue by balpert89** [ironcore-dev/ironcore-net#226]
# Summary

The `ironcore-net` stack is currently not reconciling configuration for network peering as described in [the network peering proposal]( https://github.com/ironcore-dev/ironcore/blob/main/docs/proposals/09-network-peering.md).

## Basic example

An `ironcore` network configured with a peering (assume the peered network is also available):
```
apiVersion: networking.ironcore.dev/v1alpha1
kind: Network
metadata:
  creationTimestamp: "2024-01-10T16:22:27Z"
  finalizers:
  - networking.ironcore.dev/network
  - apinet.ironcore.dev/network
  generation: 4
  name: shoot--garden-dev--fptest3
  namespace: msl0-0-shoots
  resourceVersion: "2627315"
  uid: 98f82141-1d51-4d1b-8c16-ca076db1e2ce
spec:
  peerings:
  - name: peering-fptest2
    networkRef:
      name: shoot--garden-dev--fptest2
      namespace: msl0-0-shoots
  - name: peering-fptest1
    networkRef:
      name: shoot--garden-dev--fptest1
      namespace: msl1-0-shoots
  providerID: ironcore-net://ironcore-net-system/98f82141-1d51-4d1b-8c16-ca076db1e2ce/13170443/630c5c6d-0248-4b66-a74e-5c814e1dd6a4
status:
  peerings:
  - name: peering-fptest2
  - name: peering-fptest1
  state: Available
```

The corresponding `ironcore-net` network object is missing the peering configuration:
```
apiVersion: core.apinet.ironcore.dev/v1alpha1
kind: Network
metadata:
  creationTimestamp: "2024-01-10T16:22:27Z"
  finalizers:
  - partition.metalnetlet.apinet.ironcore.dev/my-partition-name
  labels:
    apinetlet.ironcore.dev/network-name: shoot--garden-dev--fptest3
    apinetlet.ironcore.dev/network-namespace: msl0-0-shoots
    apinetlet.ironcore.dev/network-uid: 98f82141-1d51-4d1b-8c16-ca076db1e2ce
  name: 98f82141-1d51-4d1b-8c16-ca076db1e2ce
  namespace: ironcore-net-system
  resourceVersion: "502"
  uid: 630c5c6d-0248-4b66-a74e-5c814e1dd6a4
spec:
  id: "202"
status: {}
```

**Comment by afritzler** [comment]
/ref #251 describing the peering feature on a higher level.

**Comment by ushabelgur** [comment]
@afritzler I think we can close this issue as well, as #251 is closed.

**Comment by afritzler** [comment]
Fixed with #251
